src/components/ZoomableCards/ZoomCardOpened.js

```diff
@@ -0,0 +1,196 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import { Transition, animated } from 'react-spring'
+import { Viewport, springs } from '@aragon/ui'
+
+function lerp(progress, value1, value2) {
+  return (value2 - value1) * progress + value1
+}
+
+const LARGE_MARGIN = 30
+const SIDEBAR_WIDTH = 360
+
+class ZoomCardOpened extends React.Component {
+  static propTypes = {
+    cardRect: PropTypes.instanceOf(DOMRect),
+    renderContent: PropTypes.func,
+    renderAside: PropTypes.func,
+    currentId: PropTypes.string,
+  }
+
+  state = {
+    openedCardRect: null,
+  }
+
+  _openedCard = React.createRef()
+
+  componentDidMount() {
+    this.setState({ openedCardRect: this.getOpenedCardRect() })
+  }
+
+  componentDidUpdate(prevProps) {
+    this.updateCardRect(prevProps.cardRect, this.props.cardRect)
+  }
+
+  updateCardRect(prevCardRect, cardRect) {
+    if (!cardRect || cardRect === prevCardRect) {
+      return
+    }
+
+    this.setState({ cardRect, openedCardRect: this.getOpenedCardRect() })
+  }
+
+  getOpenedCardRect() {
+    return this._openedCard.current.getBoundingClientRect()
+  }
+
+  getTransform = t => {
+    const { cardRect, openedCardRect } = this.state
```

Comment:
> Can we get `cardRect` from `this.props`?

It’s initially coming from the props, but it is saved in the state (so that we remember what the rect of the closed card was when closing).

> If it does need to be in state, we should make sure to add it in the initial state, and perhaps also set it in `componentDidMount()`.

Yes totally :+1: